src/main/java/teammates/common/util/Url.java

```diff
@@ -2,12 +2,16 @@
 
 public class Url {
 
-    private String urlString;
+    protected String urlString;
 
     public Url(String url) {
-        this.urlString = url;
+        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
     }
```

Comment:
Sure the class may have worked like that all along, or the behavior diverged from original intentions at some point, but nevertheless, `Url` is not the right name for this class because the class does not represent any old URL but specifically our app URLs. Code should work 'as advertised', and this one doesn't. 
If `AppUrl` is a better name, we should go ahead and change it even if it results in widespread changes. It's just an automatic refactoring so should be easy to do. May be as a separate `forContributors` issue?